instruments/keithley/keithley580.py

```diff
@@ -248,7 +245,7 @@ def trigger_mode(self):
     def trigger_mode(self, newval):
         if isinstance(newval, str):
             newval = Keithley580.TriggerMode[newval]
-        if newval not in Keithley580.TriggerMode:
+        if not isinstance(newval, Keithley580.TriggerMode):
```

Comment:
The error message reads as if this should be an instance check. In the other properties, this is an instance check, only here something weird is tested for. Changed it to an instance check for consistency, and to make it agree with the Type Error and the message.